backend/src/chats/chats.controller.ts

```diff
@@ -115,4 +125,105 @@ export class ChatsController {
     const role = await this.chatsService.getMemberRole(id, userId);
     return { role };
   }
+
+  @Post(':id/kick')
+  @HttpCode(HttpStatus.NO_CONTENT)
+  @ChannelRoles(ChatMemberRole.OWNER, ChatMemberRole.ADMIN)
+  async kickMember(
+    @User('id') userId: number,
+    @Param('id', ParseIntPipe) id: number,
+    @Body() dto: KickMemberDto,
+  ): Promise<void> {
+    const { kickUserId } = dto;
+
+    await this.chatsService.kickMember(id, userId, dto);
+
+    this.eventEmitter.emit('chat.kick', {
+      userId: kickUserId,
+      chatId: id,
+    });
+  }
+
+  @Post(':id/mute')
+  @HttpCode(HttpStatus.NO_CONTENT)
+  @ChannelRoles(ChatMemberRole.OWNER, ChatMemberRole.ADMIN)
+  async muteMember(
+    @User('id') userId: number,
+    @Param('id', ParseIntPipe) id: number,
+    @Body() dto: MuteMemberDto,
+  ): Promise<void> {
+    await this.chatsService.muteMember(id, userId, dto);
+
+    this.eventEmitter.emit('chat.mute', {
+      userId: dto.muteUserId,
+      chatId: id,
+    });
+  }
+
+  @Post(':id/unmute')
+  @HttpCode(HttpStatus.NO_CONTENT)
+  @ChannelRoles(ChatMemberRole.OWNER, ChatMemberRole.ADMIN)
+  async unmuteMember(
+    @User('id') userId: number,
+    @Param('id', ParseIntPipe) id: number,
+    @Body() dto: UnmuteMemberDto,
+  ): Promise<void> {
+    const { unmuteUserId } = dto;
+
+    await this.chatsService.unmuteMember(id, userId, dto);
+
+    this.eventEmitter.emit('chat.unmute', {
+      userId: unmuteUserId,
+      chatId: id,
+    });
+  }
```

Comment:
O que tu acha de desestruturar o dto e passar só o que o service precisa pra função do service? O dto é legal quando a info vem de fora (front -> back) mas do controller pro service acho que dá pra simplificar assim:
```suggestion
  @Post(':id/unmute')
  @HttpCode(HttpStatus.NO_CONTENT)
  @ChannelRoles(ChatMemberRole.OWNER, ChatMemberRole.ADMIN)
  async unmuteMember(
    @User('id') userId: number,
    @Param('id', ParseIntPipe) id: number,
    @Body() { unmuteUserId }: UnmuteMemberDto,
  ): Promise<void> {
    await this.chatsService.unmuteMember(id, userId, unmuteUserId);

    this.eventEmitter.emit('chat.unmute', {
      userId: unmuteUserId,
      chatId: id,
    });
  }
```
E a função do service recebe direto o que ela vai usar:
```ts
  async unmuteMember(
    chatId: number,
    userId: number,
    unmuteUserId: number,
  ): Promise<ChatMemberEntity> {
    const chat = await this.findOne(chatId);
// ... 
```
Se achar uma boa eu topo fazer essa chore pra ficar padrão em todos :)

Reply:
eh que no caso tem serviços que usam mais de uma das informações do DTO. acho que o que dá pra fazer é padronizar os os controllers como já estão, e os serviços pra desestruturarem o que vão usar direto nos parâmetros, semelhante ao que vc fez no controller. o que acha? e acho que dá pra fazer numa PR separada tbm

Reply:
Saquei! 
O que achei estranho foi o service usar DTO
Quando tem comunicação entre partes separadas como back e front, DTO faz sentido pra garantir que o tipo que tá recebendo tá certo, como o controller que recebe coisas de fora 
Agora comunicação interna dentro do back, do controller pro service, acho que types e interfaces (o normal) é o suficiente

Reply:
Mas não tem nada de errado mesmo, por mim deixamos assim 👍